### ISSUE TITLE
Start Menu Settings Again (#145)

### ISSUE THREAD

**Issue by LiBOSS-CZ** [Disassembler0/Win10-Initial-Setup-Script#187]
I am facing issue #145 again on fresh Windows 10 Pro CZ 1809 with all updates 17763.195 (latest ISO from MS).
Script version 3.4/3.5. Same issue. 
I have also tried to reenable SharedExperiences without success.
My preset below.

[Win10-my.txt](https://github.com/Disassembler0/Win10-Initial-Setup-Script/files/2706866/Win10-my.txt)

**Comment by Disassembler0** [comment]
We have covered lot of ground in #145. I'm going to assume that you're referring to that part where you apply start menu customization tweaks (`HideRecentlyAddedApps`, `UnpinStartMenuTiles` etc.) and after a while the whole start menu resets, right?

**Comment by LiBOSS-CZ** [comment]
I know :-( Exactly. Whole Start menu resets after a short time.

**Comment by Disassembler0** [comment]
Can't replicate on clean installation of neither 17763.195 nor 17763.253.

But I saw it today one device which isn't used very often and today it was finally upgraded to 1809. Few minutes after reboot, the start menu categories and icons reset. So I went and check and sure enough there was still the old version of `DisableSharedExperiences` applied.

The new version of the tweak doesn't touch the registry keys used in the old version, so please check manually if you have `EnableCdp` and `EnableMmx` values set under `HKLM\SOFTWARE\Policies\Microsoft\Windows\System` and delete them if you do.

**Comment by LockTar** [comment]
> The new version of the tweak doesn't touch the registry keys used in the old version, so please check manually if you have `EnableCdp` and `EnableMmx` values set under `HKLM\SOFTWARE\Policies\Microsoft\Windows\System` and delete them if you do.

Checked the registry but don't see anything under it.

**Comment by LiBOSS-CZ** [comment]
Same for me. I am sorry, but It was too annoying for me, so I did clean fresh install. With this new fresh install it did not behave bad like that anymore.

**Comment by Disassembler0** [comment]
@LockTar: Can you please send a screenshot of your start menu to confirm that we're observing the same problem? Also, what's your Windows build number? (Run `winver` command to get the info if you're not sure)

**Comment by LockTar** [comment]
@Disassembler0 My version is: 1809 => 17763.253

![image](https://user-images.githubusercontent.com/1568453/51439198-0df0f980-1cb7-11e9-938b-c0bd4454254d.png)

To be honest. I'm not sure if the issue that I'm having is related with this project. I'm testing a Microsoft Intune setup for a company that I currently working for. Was working fine for months until I upgraded to 1809. Then I searched on Google for the problem and everything was pointing to this GitHub project. Also, every setting is the same as people are saying in this issue. 

Is there a way I can test if this is used in my company?

**Comment by Disassembler0** [comment]
> Is there a way I can test if this is used in my company?

Wait, so you're implying that you haven't used the script and you have no knowledge if has ever been used? That may complicate things a bit. :/

The tweaks are just a bunch of registry modifications, which mostly result in the same changes as whatever group policy templates or clicking on UI controls manually would do. I don't think there is anything what would uniquely identify that this script was used.

---

The crucial thing to say if this is the same problem or not is the menu layout resetting. If you unpin any tiles or change anything in the start menu, are the changes automatically reverted after some time?

1) You've already checked the registry keys for `EnableCdp` and `EnableMmx`, so it's safe to rule out the same cause as in #145.

2) Please check start type of *CDPUserSvc*. You can do so with following snippet
    ```powershell
    Get-Service CDPUserSvc | Select -Property StartType
    ```
    If it says anything other than *Automatic*, run
    ```powershell
    Set-Service "CDPUserSvc" -StartupType Automatic
    ```
   and reboot. This was originally related to #170, but it revolves around the same tweak and similar type of problem.

3) I have seen the "dead" tiles with downwards pointing arrows a long time ago (December 2017) in relation with disabled or broken Cortana background service. Please run this snippet to enable all background apps (incl. Cortana) and then logout and log back in to see if there will be any difference.
    ```powershell
    Remove-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\BackgroundAccessApplications" -Name "GlobalUserDisabled" -ErrorAction SilentlyContinue
    Get-ChildItem -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\BackgroundAccessApplications" | ForEach-Object {
        Remove-ItemProperty -Path $_.PsPath -Name "Disabled" -ErrorAction SilentlyContinue
        Remove-ItemProperty -Path $_.PsPath -Name "DisabledByUser" -ErrorAction SilentlyContinue
    }
    ```

**Comment by LockTar** [comment]
> Wait, so you're implying that you haven't used the script and you have no knowledge if has ever been used? That may complicate things a bit. :/

Yes that's why I'm saying that I maybe don't have the same issue. So sorry in advanced if I'm wasting your time... But everything looks like it. This company is to big to get a answer if this is used because I already tried for weeks.... But the problem started after Intune and 1809 update...

> The crucial thing to say if this is the same problem or not is the menu layout resetting. If you unpin any tiles or change anything in the start menu, are the changes automatically reverted after some time?

Yes that is exactly the thing that I have. Also tried to reset the menu from a fresh profile with winaero but that was also reset after a while...

1. The folder Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\System doesn't exist and checked "Continue experiences on this device" and is set to "Not Configured"
2. It says Automatic
3. **Done and I think this worked!!!** Have done this yesterday in the morning and now almost 24 hours later the tiles are still how I customized them. Also the menu items doesn't show "new" anymore when I opened that program.
